setup.py

```diff
@@ -12,28 +12,60 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from setuptools import setup
+import ast
+from setuptools import setup, find_packages
+
+
+def _read_me() -> str:
+    with open("README.md", "rt", encoding="utf8") as fh:
+        readme = fh.read()
+    return readme
+
+
+def _get_version() -> str:
+    # ops.__init__ needs to pull in ops.charm to work around a circular
```

Comment:
This makes me think we should have a separate 'version.py' file that doesn't import anything but gets imported by __init__.
I guess that still breaks because to import foo/version.py you have to import __init__ as well?

Reply:
yes, importing `foo.version` implicitly imports `foo`. I did play with using the import internals to do this, but ultimately punted on it as it's not a lot simpler than going the `ast` route.